Doesn't build with dlist 1.0

```
[ 6 of 21] Compiling Data.ListLike.DList ( src/Data/ListLike/DList.hs, dist/build/Data/ListLike/DList.dyn_o )

src/Data/ListLike/DList.hs:37:10: error:
    • Couldn't match type ‘[a]’ with ‘DList a’
      Expected type: DList a -> DList a
        Actual type: DList a -> [a]
    • In the expression: D.tail
      In an equation for ‘tail’: tail = D.tail
      In the instance declaration for ‘ListLike (DList a) a’
    • Relevant bindings include
        tail :: DList a -> DList a
          (bound at src/Data/ListLike/DList.hs:37:3)
   |
37 |   tail = D.tail
   |          ^^^^^^
```

--- a/ListLike.cabal
+++ b/ListLike.cabal
@@ -56,7 +56,7 @@ Library
                 ,array      >= 0.3   && < 0.6
                 ,text       >= 0.11  && < 1.3
                 ,vector     >= 0.5   && < 0.13
-                ,dlist      >= 0.7   && < 0.9
+                ,dlist      >= 0.7   && < 1.1
                 ,fmlist     >= 0.8   && < 0.10
                 ,utf8-string
                 ,deepseq

--- a/src/Data/ListLike/DList.hs
+++ b/src/Data/ListLike/DList.hs
@@ -1,5 +1,6 @@
 {-# LANGUAGE NoImplicitPrelude, MultiParamTypeClasses, FlexibleInstances #-}
 {-# OPTIONS -fno-warn-orphans #-}
+{-# LANGUAGE CPP #-}
 -- | 'Data.ListLike.ListLike' instances for 'Data.DList.DList'
 module Data.ListLike.DList () where
 --import qualified Prelude as P
@@ -34,7 +35,11 @@ instance ListLike (DList a) a where
   snoc = D.snoc
   append = D.append
   head = D.head
+#if MIN_VERSION_dlist(1,0,0)
+  tail = D.fromList . D.tail
+#else
   tail = D.tail
+#endif
   rigidMap = D.map
   null = null . D.toList
   --toList = D.toList

